crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension_in_call.rs

```diff
@@ -11,15 +11,19 @@ use crate::checkers::ast::Checker;
 use crate::rules::flake8_comprehensions::fixes;
 
 /// ## What it does
-/// Checks for unnecessary list comprehensions passed to `any` and `all`.
+/// Checks for unnecessary list comprehensions passed to builtin functions that take an iterable.
 ///
 /// ## Why is this bad?
-/// `any` and `all` take any iterators, including generators. Converting a generator to a list
-/// by way of a list comprehension is unnecessary and requires iterating all values, even if `any`
-/// or `all` could short-circuit early.
+/// Many builtin functions (this rule currently covers `any`, `all`, `min`, `max`, and `sum`) take
+/// any iterable, including a generator. Converting a generator to a list by way of a list
+/// comprehension is unnecessary and wastes memory for large iterables by fully materializing a
+/// list rather than handling values one at a time.
```

Comment:
The issue pre-existed your PR, but: list comprehensions don't really "convert generators to lists"

```suggestion
/// any iterable, including a generator. Constructing a list via a comprehension
/// only to immediately consume the list wastes memory when values could be iterated
/// over lazily via a generator expression instead.
```